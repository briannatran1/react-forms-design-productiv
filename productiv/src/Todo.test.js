import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

describe('Todo component', function () {
  const todo = { id: 1, title: 'testing', description: 'finish testing', priority: 2 };

  test('renders without crashing', function () {
    render(
      <Todo
        title={todo.title}
        description={todo.description}
        priority={todo.priority} />);
  });

  test('matches snapshot', function () {
    const { asFragment } = render(
      <Todo
        title={todo.title}
        description={todo.description}
        priority={todo.priority} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('contains expected text from todo', function () {
    const { queryByText } = render(
      // const result = render(
      <Todo
        title={todo.title}
        description={todo.description}
        priority={todo.priority} />);

    // debug(queryByText('Priority:'));
    expect(queryByText('testing')).toBeInTheDocument();
    expect(queryByText('(Priority: 2)')).toBeInTheDocument();
    expect(queryByText('finish testing')).toBeInTheDocument();
  });
});
