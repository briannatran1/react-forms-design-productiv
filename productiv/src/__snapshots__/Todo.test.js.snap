// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <div>
      <b>
        testing
      </b>
       
      <small>
        (Priority: 2)
      </small>
    </div>
    <div>
      <small>
        finish testing
      </small>
    </div>
  </div>
</DocumentFragment>
`;
